### PR TITLE
1766 add lower limit and upper limit for time series forecasting

### LINKED ISSUE #1766: Add lower limit and upper limit for time series forecasting
Time series data can be naturally clamped at a limit - could be lower limit or upper limit. 
Provide an option to the user to set this limit such that the predictions and intervals do not cross it.

Reference [Link](https://www.linkedin.com/feed/update/urn:li:activity:6858135434724159488?commentUrn=urn%3Ali%3Acomment%3A%28activity%3A6858135434724159488%2C6858178529432100864%29&replyUrn=urn%3Ali%3Acomment%3A%28activity%3A6858135434724159488%2C6858189510220541952%29)

### PR DESCRIPTION
**Pull request by AJarman**
## Related Issuse or bug

Info about Issue or bug

Closes #1766 

#### Describe the changes you've made

A clear and concise description of what you have done to successfully close your assigned issue. Any new files? or anything you feel to let us know!

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)

A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

## Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot**  | <b>updated screenshot </b> |

### CODE REVIEW

File: pycaret/internal/preprocess/time_series/forecasting/preprocessor.py [hunk 1]
```diff
@@ -152,11 +157,13 @@ def _add_transformation_steps(self, transform: str, target: bool = True):
...
-                "log": LogTransformer(),
-                "sqrt": SqrtTransformer(),
-                "exp": ExponentTransformer(),
-                "cos": CosineTransformer(),
+                "box-cox": BoxCoxTransformer,
+                "log": LogTransformer,
+                "sqrt": SqrtTransformer,
+                "exp": ExponentTransformer,
+                "cos": CosineTransformer,
+                "scaledlogit" : ScaledLogitTransformer
```
Comment (ngupta23) [inline pycaret/internal/preprocess/time_series/forecasting/preprocessor.py:165]: I think the `ScaledLogitTransformer` should be done after imputation and separate from other transformations

So someone might want to limit the data but also apply log/box-cox transformations if residual variance needs to be satisfied.

What do you think?

File: pycaret/internal/preprocess/time_series/forecasting/preprocessor.py [hunk 2]
```diff
@@ -166,15 +173,27 @@ def _add_transformation_steps(self, transform: str, target: bool = True):
...
         else:
             raise ValueError(
-                f"{type_} transformaton method '{type(transform)}' is not of allowed type."
+                f"{type_} transformaton method '{type(transform)}' is not of allowed type (str)."
             )
 
         if target:
-            transformer = transform_dict[transform]
-            self.pipe_steps_target.extend([("transformer", transformer)])
+            if self.transform_target_kwargs:
```
Comment (ngupta23) [inline pycaret/internal/preprocess/time_series/forecasting/preprocessor.py:180]: you can just replace this with

```python
trans_kwargs = self.transform_exogenous_kwargs or {}
```

File: pycaret/time_series/forecasting/oop.py [hunk 3]
```diff
@@ -1075,7 +1075,9 @@ def setup(
         numeric_imputation_target: Optional[Union[int, float, str]] = None,
         numeric_imputation_exogenous: Optional[Union[int, float, str]] = None,
         transform_target: Optional[str] = None,
+        transform_target_kwargs: Optional[dict] = None,
```
Comment (ngupta23) [inline pycaret/time_series/forecasting/oop.py:1078]: I will have to check with the rest of the team if we want to add kwargs or dedicated arguments.

Things to consider

1. Consistency with the previous interface for regression/classification.
2. We also plan to add a custom pipeline option as in regression/classification so the kwargs may be a moot point if we provide some dedicated arguments (i.e. users can specify their custom pipeline if we want to customize beyond what we provide).

I will try to get you an answer by the weekend.

Comment (ngupta23) [reply pycaret/time_series/forecasting/oop.py:1078]: @AJarman : Interface feedback left here: https://github.com/pycaret/pycaret/issues/1766#issuecomment-1094078302

**Comment by AJarman** [comment]
Refactoring as per comments above

**Pull request closed without merging**